exporter/exporterhelper/internal/queuebatch/config.go

```diff
@@ -79,10 +79,16 @@ func (cfg *Config) Validate() error {
 		return errors.New("`wait_for_result` is not supported with a persistent queue configured with `storage`")
 	}
 
-	if cfg.Batch.HasValue() && cfg.Batch.Get().Sizer == cfg.Sizer {
-		// Avoid situations where the queue is not able to hold any data.
-		if cfg.Batch.Get().MinSize > cfg.QueueSize {
-			return errors.New("`min_size` must be less than or equal to `queue_size`")
+	if cfg.Batch.HasValue() {
+		if err := cfg.Batch.Get().Validate(); err != nil {
+			return err
+		}
```

Comment:
Can we change https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/xconfmap/config.go#L28 to call this for us?

Reply:
The reason is that we use that in Service and ask everyone to not manually call Validate for embedded fields. So I think this should work for this case as well.

Reply:
Well, now I see that xconfmap is where this code lives (thought it was confmap).
https://github.com/open-telemetry/opentelemetry-collector/issues/13579#issuecomment-3161870006